awspub/sns.py

```diff
@@ -0,0 +1,81 @@
+"""
+Methods used to handle notifications for AWS using SNS
+"""
+
+import json
+import logging
+from typing import Any, Dict, List
+
+import boto3
+from mypy_boto3_sns.client import SNSClient
+from mypy_boto3_sts.client import STSClient
+
+from awspub.context import Context
+from awspub.exceptions import AWSNotificationException
+
+logger = logging.getLogger(__name__)
+
+
+class SNSNotification(object):
+    """
+    A data object that contains validation logic and
+    structuring rules for SNS notification JSON
+    """
+
+    def __init__(self, context: Context, image_name: str, region_name: str):
```

Comment:
this could be a `dataclass ` since we don't do anything custom in the constructor

Reply:
we don't use dataclasses in the other classes yet. For consistency, I would keep it the way it currently is.

Reply:
i'm not sure i follow the argument for this case. if i just want a simple data container would you still recommend using `__init__` for now?

Reply:
yes.